echopype/echodata/simrad.py

```diff
@@ -139,16 +142,9 @@ def _retrieve_correct_beam_group_EK80(
         # 3) power samples are in Sonar/Beam_group2 if two beam groups exist
 
         # Raise error if waveform_mode="CW" but CW data does not exist
-        if (
-            encode_mode == "complex"  # only check if encode_mode="complex"
-            and "frequency_start" in echodata["Sonar/Beam_group1"]  # only check is data is BB
-        ):
-            if (
-                echodata["Sonar/Beam_group1"]["channel"].size  # total number of channels
-                == echodata["Sonar/Beam_group1"]["frequency_start"]
-                .dropna(dim="channel")["channel"]
-                .size  # number of BB channel
-            ):  # if all channels are BB
+        if encode_mode == "complex" and transmit_type_has_BB:
+            # complex + BB data (all channels are BB)
+            if np.all(echodata["Sonar/Beam_group1"]["transmit_type"] != "CW"):
                 raise ValueError("waveform_mode='CW', but all data are broadband (BB)!")
```

Comment:
I think there is a logic error in the first if here. `transmit_type_has_BB` says whether or not there is one ping that is not CW, but does not tell us whether there is no CW ping at all (which is the case we want to catch, and is in the second if. I think this can be:
```python
        # Raise error if waveform_mode="CW" but CW data does not exist (not a single ping is CW)
        if encode_mode == "complex" and np.all(echodata["Sonar/Beam_group1"]["transmit_type"] != "CW"):
            raise ValueError("waveform_mode='CW', but all data are broadband (BB)!")
```

Reply:
I think it's not so much that the logic is in error (or led to errors), but rather that having `transmit_type_has_BB` in the first if is unnecessary and doesn't add anything useful. I now see that I translated the original logic too literally, but didn't take the further step you did. I agree with your suggestion and will implement it. Thanks! FYI, I'll also remove the variable `transmit_type_has_BB` since it will no longer serve the purpose of reusability.

Reply:
This change is in the latest commits.